quark.c

```diff
@@ -2897,3 +2898,42 @@ quark_queue_get_event(struct quark_queue *qq)
 
 	return (qev);
 }
+
+int
+quark_start_kube_talker(const char *kube_config, pid_t *pid)
+{
+        int     pipefd[2];
+        char    pidbuf[16];
+
+        if (kube_config == NULL || pid == NULL)
+                return (errno = EINVAL, -1);
+
+        *pid = -1;
+        if ((pipe(pipefd)) == -1)
+                err(1, "pipe");
+        if ((*pid = fork()) == -1)
+                err(1, "fork");
+
+        /* parent */
+        if (*pid != 0) {
+                close(pipefd[1]);
+
+                return (pipefd[0]);
+        }
+
+        /* child */
+        close(pipefd[0]);
+
+        if (qclosefrom(3, pipefd[1]) == -1) {
+                qwarn("qclosefrom");
+                qwarnx("not closing child descriptors");
+        }
+
+        snprintf(pidbuf, sizeof(pidbuf), "%d", pipefd[1]);
```

Comment:
why named `pidbuf`? it's filled with a file descriptor, right?

Reply:
whoooooooooooooops, it used to be a pid, and the pipe was on another buffer, I changed and renamed the wrong one, I'll fix it, I also realized, that I had my tabs disabled in emacs and the whole function is not using tabs. :D